editor/sidebar/post-schedule/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import clickOutside from 'react-click-outside';
+import DatePicker from 'react-datepicker';
+import moment from 'moment';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+import { Component } from 'element';
+import { dateI18n, settings } from 'date';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import PostScheduleClock from './clock';
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+class PostSchedule extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			open: false,
+		};
+		this.toggleDialog = this.toggleDialog.bind( this );
+	}
+
+	toggleDialog( event ) {
+		event.preventDefault();
+		this.setState( { opened: ! this.state.opened } );
+	}
+
+	handleClickOutside() {
+		this.setState( { opened: false } );
+	}
+
+	render() {
+		const { date, onUpdateDate } = this.props;
+		const momentDate = date ? moment( date ) : moment();
+		const label = date
+			? dateI18n( settings.formats.datetime, date )
+			: __( 'Immediately' );
+		const handleChange = ( newDate ) => {
+			onUpdateDate( newDate.format( 'YYYY-MM-DDTHH:mm:ss' ) );
+		};
+		const is12HourTime = settings.formats.time.indexOf( 'a' ) !== -1 || settings.formats.time.indexOf( 'A' ) !== -1;
```

Comment:
Even this solution is not perfect because this `\\\\\a` is a 12 hour time but the test returns `false`